test/crash/src/test/java/io/narayana/lra/arquillian/LRACoordinatorRecoveryIT.java

```diff
@@ -217,14 +205,12 @@ public void lraCoordinatorRecoveryTwoLRAs(@ArquillianResource @OperateOnDeployme
         // Restarts lra-coordinator to simulate a crash
         restartContainer(LRA_COORDINATOR_CONTAINER_QUALIFIER);
 
-        // Waits for a period of time longer than the timeout of the short LRA transaction
-        doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
-        int sc = recover();
-
-        if (sc != 0) {
-            recover();
-        }
+        // Waits for a period of time longer than the timeout of the short LRA
+        // transaction
+        doWait(SHORT_TIMEOUT + 1000);
+        // the recovery is started during the server restart asynchronously, so calling
+        // a synchronous recovery to make sure the recovery is finished
+        while (recover() != 0);
```

Comment:
Does this definitely need to be in a while loop? I would assume a single call to recover() should complete the in-doubt LRA?

Reply:
You are right @tomjenkinson, I have updated the test so that the recover is made after the time limit is reached. Thanks